src/Omega_h_tag.hpp

```diff
@@ -1,35 +1,84 @@
 #ifndef OMEGA_H_TAG_HPP
 #define OMEGA_H_TAG_HPP
 
+#include <unordered_map>
 #include <Omega_h_array.hpp>
+#ifdef OMEGA_H_USE_MPI
+#include <mpi.h>
+#endif
 
 namespace Omega_h {
 
 inline void check_tag_name(std::string const& name) {
   OMEGA_H_CHECK(!name.empty());
 }
 
+enum class ArrayType {
+  NotSpecified,
+  VectorND, // vector with N components
+  SymmetricSquareMatrix, // symmetric matrix with dim*(dim+1)/2 components
+  FullSquareMatrix , // full matrix with dim*dim components
```

Comment:
It doesn't look like this is used in the code, and based on prior discussions, I think we should remove it until there is a need for it.